src/logging.rs

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public

```

Comment:
I put this logic in a new file because it did not seem to fit in anywhere else. I'd be happy to move it elsewhere tho.